packages/components/tabs/src/TabsList.tsx

```diff
@@ -0,0 +1,183 @@
+import * as RadixTabs from '@radix-ui/react-tabs'
+import { Button } from '@spark-ui/button'
+import { Icon } from '@spark-ui/icon'
+import { ArrowVerticalLeft } from '@spark-ui/icons/dist/icons/ArrowVerticalLeft'
+import { ArrowVerticalRight } from '@spark-ui/icons/dist/icons/ArrowVerticalRight'
+import { forwardRef, type ReactElement, useCallback, useEffect, useRef, useState } from 'react'
+
+import { useTabsContext } from './TabsContext'
+import { listStyles, navigationArrowStyles, wrapperStyles } from './TabsList.styles'
+import { useResizeObserver } from './useResizeObserver'
+
+export interface TabsListProps extends Omit<RadixTabs.TabsListProps, 'children'> {
+  /**
+   * Change the component to the HTML tag or custom component of the only child. This will merge the original component props with the props of the supplied element/component and change the underlying DOM node.
+   * @default false
+   */
+  asChild?: boolean
+  /**
+   * When true, keyboard navigation will loop from last tab to first, and vice versa.
+   * @default false
+   */
+  loop?: boolean
+  children: ReactElement[]
+}
+
+export const TabsList = forwardRef<HTMLDivElement, TabsListProps>(
+  (
+    {
+      /**
+       * Default Radix Primitive values
+       * see https://www.radix-ui.com/docs/primitives/components/tabs#list
+       */
+      asChild = false,
+      loop = false,
+      children,
+      className,
+      ...rest
+    },
+    ref
+  ) => {
+    const wrapperRef = useRef(null)
+    const innerRef = useRef(null)
+    const listRef = ref || innerRef
+    const { orientation } = useTabsContext()
+
+    const { width } = useResizeObserver(wrapperRef)
+
+    const [withArrows, setWithArrows] = useState<
+      Record<'prev' | 'next', 'visible' | 'hidden' | 'disabled'>
+    >({ prev: 'hidden', next: 'hidden' })
+
+    useEffect(() => {
+      /**
+       * Show/hide arrows
+       */
+      /* istanbul ignore if -- @preserve */
+      if (!('current' in listRef) || !listRef.current) {
+        return
+      }
+
+      if (orientation !== 'horizontal') {
+        setWithArrows({ prev: 'hidden', next: 'hidden' })
+      } else {
+        setWithArrows({
+          prev: listRef.current.scrollWidth > listRef.current.clientWidth ? 'visible' : 'hidden',
+          next: listRef.current.scrollWidth > listRef.current.clientWidth ? 'visible' : 'hidden',
+        })
+      }
+    }, [orientation, listRef, width])
+
+    useEffect(() => {
+      /**
+       * Enable/disable arrows
+       */
+      /* istanbul ignore if -- @preserve */
+      if (!('current' in listRef) || !listRef.current || withArrows.prev === 'hidden' || loop) {
+        return
+      }
+
+      const toggleArrowsVisibility = (target: HTMLDivElement) => {
+        setWithArrows({
+          prev: target.scrollLeft > 0 ? 'visible' : 'disabled',
+          next:
+            target.scrollLeft + target.clientWidth < target.scrollWidth ? 'visible' : 'disabled',
+        })
+      }
```

Comment:
```suggestion
      const toggleArrowsVisibility = (event: Event) => {
       if (!(event.target instanceof HTMLDivElement)) return
       const { target } = event
        setWithArrows({
          prev: target.scrollLeft > 0 ? 'visible' : 'disabled',
          next:
            target.scrollLeft + target.clientWidth < target.scrollWidth ? 'visible' : 'disabled',
        })
      }
```

Reply:
Could you please explain the win in this change?